vsphere/datadog_checks/vsphere/data/conf.yaml.example

```diff
@@ -82,49 +60,63 @@ instances:
+    ## 'patterns' is a list of matching regex, if any of those matches, the resource will be monitored.
+    ## 'property' is the kind of property on which to apply the filter and must be one of those values:
+    ##    - 'property: name' (default) to filter on the resource name (defined by vCenter)
+    ##    - 'property: inventory_path' to filter on the full inventory path
+    ##       (ex: \/<Datacenter_name>\/vm\/<FOLDER_NAME>\/.* for all vms in the folder <FOLDER_NAME>)
+    ##    - 'property: hostname' to filter on the ESXi Host name (only valid for if 'resource' equals 'vm')
+    ##    - 'property: guest_hostname' to filter on the VM hostname (only valid for if 'resource' equals 'vm' and
+    ##       if VMware tools are installed)
+    ## Note1: 'datastore', 'datacenter' and 'cluster' filters are ignored when collecting realtime metrics as those
+    ##         resources do not have realtime metrics
+    ## Note2: 'vm' and 'host' filters are ignored when collecting historical metrics as those resources do not have
+    ##         historical metrics
+    #
+    # resource_filters:
+    #   - resource: vm
+    #     property: name
+    #     patterns:
+    #       - <VM_REGEX_1>
+    #       - <VM_REGEX_2>
+    #   - resource: vm
+    #     property: hostname
+    #     patterns:
+    #       - <HOSTNAME_REGEX>
+    #   - resource: vm
+    #     property: guest_hostname
+    #     patterns:
+    #       - <GUEST_HOSTNAME_REGEX>
+    #   - resource: host
+    #     property: inventory_path
+    #     patterns:
+    #       - <INVENTORY_PATH_REGEX>
+
+    ## @param metric_filters - object - optional - default: all metrics
+    ## For each resource type (vm, host, datastore, datacenter, cluster) collected,
+    ## you can choose which metric you want to collect using a list of regex.
+    ## If you do not specify a regex for the resource, all metrics will be collected.
+    ## See https://github.com/DataDog/integrations-core/blob/master/vsphere/metadata.csv
+    ## for the list of collected metrics.
+    ## Note1: 'datastore', 'datacenter' and 'cluster' filters are ignored when collecting realtime metrics as those
+    ##         resources do not have realtime metrics
+    ## Note2: 'vm' and 'host' filters are ignored when collecting historical metrics as those resources do no
+    ##         have historical metrics
+    #
+    # metric_filters:
+    #   vm:
+    #     - <VM_REGEX>                 # Only possible with "collection_type: realtime"
+    #   host:
+    #     - <HOST_REGEX>               # Only possible with "collection_type: realtime"
+    #   datastore:
+    #     - <DATASTORE_REGEX>          # Only possible with "collection_type: historical"
+    #   datacenter:
+    #     - <DATACENTER_REGEX>         # Only possible with "collection_type: historical"
+    #   cluster:
+    #     - <CLUSTER_REGEX>            # Only possible with "collection_type: historical"
```

Comment:
Is there a need to provide both blacklist_patterns and whitelist_pattern for `metric_filters`?

Reply:
For consistency with other integrations yes

Reply:
I think it's makes the code harder is not a requirement for people to achieve what they need. I'd suggest adding this in a future PR when we see the need for it. Wyt?